src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs

```diff
@@ -4,92 +4,236 @@
+                            buildBasePathOption));
+                }
+                else
+                {
+                    config.OnExecute(() => Run(assemblyNamesArgument, protocolOption));
+                }
             });
         }
 
-        public static ProjectContext ResolveProjectContext(string projectFilePath)
+        private static int Dispatch(
+            CommandArgument projectArgument,
+            CommandArgument assemblyNameArgument,
+            CommandOption protocolOption,
+            CommandOption frameworkOption,
+            CommandOption configurationOption,
+            CommandOption buildBasePathOption)
         {
-            var projectContexts = ProjectContext.CreateContextForEachFramework(projectFilePath);
-            var projectContext = projectContexts.FirstOrDefault();
+            var projectFilePath = projectArgument.Value;
+            var projectFile = ProjectReader.GetProject(projectFilePath);
+            var targetFrameworks = projectFile
+                .GetTargetFrameworks()
+                .Select(frameworkInformation => frameworkInformation.FrameworkName);
 
-            if (projectContext == null)
+            NuGetFramework framework;
+            if (!TryResolveFramework(frameworkOption, targetFrameworks, projectFilePath, out framework))
             {
-                throw new InvalidOperationException(
-                    string.Format(CultureInfo.CurrentCulture, Resources.InvalidProjectFile, projectFilePath));
+                // Could not resolve framework for dispatch. Error was reported, exit early.
+                return 0;
             }
 
-            return projectContext;
+            var configurationValue = configurationOption.Value() ?? Constants.DefaultConfiguration;
+            var projectContext = new ProjectContextBuilder()
+                .WithProject(projectFile)
+                .WithTargetFramework(framework)
+                .Build();
+            var buildBasePathValue = buildBasePathOption.Value();
+            var commandFactory = new ProjectDependenciesCommandFactory(
+                projectContext.TargetFramework,
+                configurationValue,
+                outputPath: null,
+                buildBasePath: buildBasePathValue,
+                projectDirectory: projectContext.ProjectDirectory);
+
+            var dispatchArgs = new List<string>
+            {
+                CommandName,
+                assemblyNameArgument.Value,
+                "--no-dispatch"
+            };
+
+            if (protocolOption.HasValue())
+            {
+                dispatchArgs.Add("--protocol");
+                dispatchArgs.Add(protocolOption.Value());
+            }
+
+            using (var errorWriter = new StringWriter())
+            {
+                var commandExitCode = commandFactory
+                    .Create(
+                        PlatformServices.Default.Application.ApplicationName,
```

Comment:
May need to change, dependency on the outcome of https://github.com/dotnet/cli/issues/2592